src/main/java/br/com/deveficiente/calendario/controller/validacao/VerificaConflitoHorarioEvento.java

```diff
@@ -0,0 +1,32 @@
+package br.com.deveficiente.calendario.controller.validacao;
+
+import br.com.deveficiente.calendario.controller.form.EventoForm;
+import br.com.deveficiente.calendario.model.Evento;
+import br.com.deveficiente.calendario.repository.EventoRepository;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+import java.util.List;
+
+public class VerificaConflitoHorarioEvento implements Validator {
+    private EventoRepository eventoRepository;
+
+    public VerificaConflitoHorarioEvento(EventoRepository eventoRepository) {
+        this.eventoRepository = eventoRepository;
+    }
+
+    @Override
+    public boolean supports(Class<?> aClass) {
+        return EventoForm.class.isAssignableFrom(aClass);
+    }
+
+    @Override
+    public void validate(Object o, Errors errors) {
+        EventoForm eventoForm = (EventoForm) o;
+        List<Evento> eventos = eventoRepository.findEventosConflituosos(eventoForm.getInicio(), eventoForm.getFim());
+        if(eventos!=null && !eventos.isEmpty()){
```

Comment:
*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

Não tem chance do retorno aqui ser nulo. O Spring Data JPA nunca te retorna nulo :). Ele segue a regra *Não retornamos nulo dentro das regras da aplicação. Pense que seu computador vai explodir.* :P